lib/Firtool/Firtool.cpp

```diff
@@ -306,6 +306,8 @@ LogicalResult firtool::populateHWToSV(mlir::PassManager &pm,
                                       const FirtoolOptions &opt) {
   pm.nestAny().addPass(verif::createStripContractsPass());
   pm.addPass(verif::createLowerFormalToHWPass());
+  pm.addPass(
```

Comment:
Just making sure I understand this right, does `opt.shouldExtractTestCode()` state that test code should be lowered or should be removed? If it means that it should be lowered, shouldn't this pass only be run if we want to generate "formal verification" SV and symbolic values should be outright removed otherwise?

Reply:
ExtractTestCode is a FIRRTL-specific pass that moves assert/assume/cover statements into a separate Verilog file/module that can then be bound into the original location. As long as we create any asserts/assumes/covers before that pass runs, everything should be good 👍